File: app/src/main/java/com/taipei/yanghaobo/kunu/ui/detail/DogDetailViewModel.java
package com.taipei.yanghaobo.kunu.ui.detail;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.taipei.yanghaobo.kunu.data.KunuRepository;
import com.taipei.yanghaobo.kunu.db.DogEntry;

/**
 *  狗狗完整資料 VM
 */
public class DogDetailViewModel extends ViewModel {

    private LiveData<DogEntry> mDogEntryLiveData;

    /**
     * @param kunuRepository
     *          Repository
     * @param id
     *          要查詢的狗狗 id
     */
    public DogDetailViewModel(KunuRepository kunuRepository, int id) {
        mDogEntryLiveData = kunuRepository.getOneDog(id);
    }

    public LiveData<DogEntry> getOneDog() {
        return mDogEntryLiveData;
    }

}

File: app/src/main/java/com/taipei/yanghaobo/kunu/ui/detail/DogDetailViewModel.kt
package com.taipei.yanghaobo.kunu.ui.detail

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel

import com.taipei.yanghaobo.kunu.data.KunuRepository
import com.taipei.yanghaobo.kunu.db.DogEntry

/**
 * 狗狗完整資料 VM
 */
class DogDetailViewModel
/**
 * @param kunuRepository
 * Repository
 * @param id
 * 要查詢的狗狗 id
 */
  (kunuRepository: KunuRepository, id: Int) : ViewModel() {

  val oneDog: LiveData<DogEntry>

  init {
    oneDog = kunuRepository.getOneDog(id)
  }

}
